ocaml/parsing/jane_syntax_parsing.mli

```diff
@@ -169,16 +169,33 @@ end
     adding these lazily as we need them. When you add another one, make
     sure also to add special handling in [Ast_iterator] and [Ast_mapper]. *)
 
-module Expression     : AST with type ast      = Parsetree.expression
-                             and type ast_desc = Parsetree.expression_desc
-module Pattern        : AST with type ast      = Parsetree.pattern
-                             and type ast_desc = Parsetree.pattern_desc
-module Module_type    : AST with type ast      = Parsetree.module_type
-                             and type ast_desc = Parsetree.module_type_desc
-module Signature_item : AST with type ast      = Parsetree.signature_item
-                             and type ast_desc = Parsetree.signature_item_desc
-module Structure_item : AST with type ast      = Parsetree.structure_item
-                             and type ast_desc = Parsetree.structure_item_desc
+module Core_type : AST
+  with type ast      = Parsetree.core_type
+   and type ast_desc = Parsetree.core_type_desc
+
+module Constructor_argument : AST
```

Comment:
Yeah, I think it probably the best outcome?  I left it unmatched on because the design isn't 100% obvious and I didn't think you were likely to use it

Reply:
OK. Let's see how this evolves.